app/src/main/java/ch/epfl/sdp/db/AppDatabase.java

```diff
@@ -0,0 +1,106 @@
+package ch.epfl.sdp.db;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.room.Database;
+import androidx.room.Room;
+import androidx.room.RoomDatabase;
+import androidx.sqlite.db.SupportSQLiteDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@Database(entities = {LeaderboardEntity.class}, version = 1, exportSchema = false)
+abstract class AppDatabase extends RoomDatabase {
+
+    // marking the instance as volatile to ensure atomic access to the variable
+    private static volatile AppDatabase INSTANCE;
```

Comment:
Its hard to understand what it does. Can you put a javadoc so we can understand it exactly what it does?